src/data_collection.py

```diff
@@ -269,6 +269,98 @@ def print_individual_in_table(file_name):
+            if author in data_dict:
+                # condition passed, adds one to the number of commits
+                data_dict[author]["COMMITS"] += 1
+            else:
+                # condition fails, creates a new key and adds empty data
+                data_dict[author] = {
+                    "COMMITS": 1,
+                    "COMMITS_TO_TESTING": 0,
+                    "COMMITS_ELSEWHERE": 0,
+                    "PERCENT_TO_TESTING": 0,
+                    "PERCENT_NOT_TO_TESTING": 0,
+                }
+
+            count = 0  # when 0 the current commit has no changes to tests
+            for filepath in filepaths:
+                if (
+                    count is 0
+                ):  # when 0, current commit no testing changes; if not 0, this commit already had testing changes
+                    if filepath:
+                        if "test" in filepath:
+                            data_dict[author][
+                                "COMMITS_TO_TESTING"
+                            ] += 1  # the current commit has a change to testing
+                            count = 1  # found a change to testing for this commit
+                        else:
+                            pass
+                else:
+                    pass
+            # Sort list to ensure consistency when testing
+        # iterate through the data to do final calculations
+        for key in data_dict:
+            data_dict[key]["PERCENT_TO_TESTING"] = (
+                data_dict[key]["COMMITS_TO_TESTING"] / data_dict[key]["COMMITS"]
+            ) * 100
+
+            data_dict[key]["COMMITS_ELSEWHERE"] = (
+                data_dict[key]["COMMITS"] - data_dict[key]["COMMITS_TO_TESTING"]
+            )
+
+            data_dict[key]["PERCENT_NOT_TO_TESTING"] = (
+                data_dict[key]["COMMITS_ELSEWHERE"] / data_dict[key]["COMMITS"]
+            ) * 100
+        # Reformat the dictionary as a value of the key INDIVIDUAL_METRICS
+        testing_dict = {"TESTING_DICT": data_dict}
+        return testing_dict
+    return {}
+
+
+def print_testing_in_table(file_name):
+    """Create and print the table of testing data using prettytable."""
+    data_table = PrettyTable()
+    current_data = json_handler.get_dict_from_json_file(file_name)
+    dictionary = current_data["TESTING_DICT"]
+    headings = [
+        "Username",
+        "Commits",
+        "Commits to Testing",
+        "Commits Elsewhere",
+        "Percent of Commits to Testing",
+        "Percent of Commits not to Testing",
+    ]
+    data_table.field_names = headings
+    for key in dictionary:
+        data_table.add_row(
+            [
+                key,
+                dictionary[key]["COMMITS"],
+                dictionary[key]["COMMITS_TO_TESTING"],
+                dictionary[key]["COMMITS_ELSEWHERE"],
+                dictionary[key]["PERCENT_TO_TESTING"],
+                dictionary[key]["PERCENT_NOT_TO_TESTING"],
+            ]
+        )
```

Comment:
After the incoming changes, this function will no longer be needed because the new function will receive parameters regarding which categories to display so there will not be a need to write another printing function.